src/js/plugins/plugin.search.js

```diff
@@ -0,0 +1,599 @@
+    if (this.enableSearch) {
+      $el.find('.BRmobileMenu__moreInfoRow').after($(
+        `<li>
+          <span>
+            <span class="DrawerIconWrapper">
+              <img class="DrawerIcon" src="${this.imagesBaseURL}icon_search_button_blue.svg" alt="info-speaker"/>
+            </span>
+            Search
+          </span>
+          <div>
+            <form class="BRbooksearch mobile">
+              <input type="search" class="BRsearchInput" placeholder="Search inside"/>
+              <button type="submit" class="BRsearchSubmit"></button>
+            </form>
+            <div class="BRmobileSearchResultWrapper">Enter your search above.</div>
+          </div>
+        </li>`
+      ));
+    }
+    return $el;
+  };
+})(BookReader.prototype.buildMobileDrawerElement);
+
+/** @override */
+BookReader.prototype.buildToolbarElement = (function (super_) {
+  return function () {
+    const $el = super_.call(this);
+    if (this.enableSearch) {
+      const $BRtoolbarSectionSearch = $(
+        `<span class="BRtoolbarSection BRtoolbarSectionSearch">
+          <form class="BRbooksearch desktop">
+            <input type="search" class="BRsearchInput" val="" placeholder="Search inside"/>
+            <button type="submit" class="BRsearchSubmit">
+              <img src="${this.imagesBaseURL}icon_search_button.svg" />
+            </button>
+          </form>
+        </span>`
+      );
+      $BRtoolbarSectionSearch.insertAfter($el.find('.BRtoolbarSectionInfo'));
+    }
+    return $el;
+  };
+})(BookReader.prototype.buildToolbarElement);
+
+/** @override */
+BookReader.prototype.initToolbar = (function (super_) {
+  return function (mode, ui) {
+    super_.apply(this, arguments);
+
+    const onSubmitDesktop = (e) => {
+      e.preventDefault();
+      const val = $(e.target).find('.BRsearchInput').val();
+      if (!val.length) return false;
+      this.search(val);
+      return false;
+    };
+
+    const onSubmitMobile = (e) => {
+      e.preventDefault();
+      const val = $(e.target).find('.BRsearchInput').val();
+      if (!val.length) return false;
+      this.search(val, { disablePopup: true });
+      this.$('.BRmobileSearchResultWrapper').append(
+        `<div>Your search results will appear below.</div>
+          <div class="loader tc mt20"></div>`
+      );
+      return false;
+    };
+
+    // Bind search forms
+    this.$('.BRbooksearch.desktop').submit(onSubmitDesktop);
+    this.$('.BRbooksearch.mobile').submit(onSubmitMobile);
+
+    // Handle clearing the search results
+    this.$(".BRsearchInput").bind('input propertychange', () => {
+      if (this.value === "") this.removeSearchResults();
+    });
+  };
+})(BookReader.prototype.initToolbar);
+
+/**
+ * Search Options
+ * @typedef {object} searchOptions
+ * @property {boolean} goToFirstResult
+ * @property {boolean} disablePopup
+ * @property {(null|function)} error - @deprecated at v.5.0
+ * @property {(null|function)} success - @deprecated at v.5.0
+ */
```

Comment:
new `@typedef`: Search Options
We pass this around to success & error callbacks, so it was helpful to make one for it.
I abstracted it up to the "top level," vscode couldn't back link the references to this when I added it right above the callback inside the function